tests/test_1.py

```diff
@@ -0,0 +1,120 @@
+    'kingdom_2':  [-3, -3, -4, -5, -4, -3, -2, -2, -2, -1, 0, 1, 1, 1, 0, 1, 2, 1, 0, 0, 0, 1, 2, 3, 2, 2],
+    'kingdom_3':  [-5, -5],
+    'kingdom_4':  [1],
+    'kingdom_5':  [1, 2, 1, 2, 1],
+    'kingdom_6':  [1, -2, 1, -2, 1],
+    'kingdom_7':  [1, 2, 1, 1, 1],
+    'kingdom_8':  [1, -2, 1, 1, 1],
+    'kingdom_9':  [1, 1, 1, 2, 1],
+    'kingdom_10': [1, 1, 1, -2, 1],
+    'kingdom_11': [1, 1, 1, 1, 2],
+    'kingdom_12': [2, 1, 1, 1, 1],
+    'kingdom_13': [1, 1, 1, 1, -2],
+    'kingdom_14': [-2, 1, 1, 1, 1],
+    'kingdom_15': [1, 1, 2, 1, 1],
+    'kingdom_16': [1, 1, -2, 1, 1],
+
+}
+EXPECTED_OUTPUT = {
+    'kingdom_1':  9,
+    'kingdom_2':  8,
+    'kingdom_3':  1,
+    'kingdom_4':  1,
+    'kingdom_5':  5,
+    'kingdom_6':  5,
+    'kingdom_7':  3,
+    'kingdom_8':  3,
+    'kingdom_9':  3,
+    'kingdom_10': 3,
+    'kingdom_11': 2,
+    'kingdom_12': 2,
+    'kingdom_13': 2,
+    'kingdom_14': 2,
+    'kingdom_15': 3,
+    'kingdom_16': 3,
+}
+
+
+def castles_number(heights):
+    start_height, current_height = None, None
+    castles = 0
+
+    if len(heights) == 0:
+        raise Exception
+    if len(heights) == 1:
+        return 1
+
+    for i in range(len(heights)):
+        if current_height is None:
+            current_height = heights[i]
+
+        if heights[i] != current_height:
+            if (start_height is None) or (heights[i] < current_height and start_height < current_height) or \
+                    (heights[i] > current_height and start_height > current_height):
+                castles += 1
+                start_height = heights[i - 1]
+                current_height = heights[i]
+            elif (heights[i] < current_height and start_height >= current_height) or \
+                    (heights[i] > current_height and start_height <= current_height):
+                start_height = heights[i - 1]
+                current_height = heights[i]
+
+        if heights[i] == current_height and i == len(heights) - 1:
+            castles += 1
+
+    return castles
+
+
+def get_castles_number(border):
+    if len(border) == 0:
+        raise Exception
+    if len(set(border)) == 1:
+        return 1
+
+    castles = 2
+    curr_direction, prev_direction = None, None
+
+    for i in range(1, len(border)):
+        if border[i] > border[i - 1]:
+            curr_direction = 1
+        elif border[i] < border[i - 1]:
+            curr_direction = -1
+
+        if prev_direction is None:
+            prev_direction = curr_direction
+            continue
+
+        if curr_direction + prev_direction == 0:
+            castles += 1
+            prev_direction = curr_direction
+
+    return castles
+
+
+class TestTask(unittest.TestCase):
+
+    def test_1(self):
+        for key, value in BORDER_HEIGHT.items():
+            self.assertEqual(EXPECTED_OUTPUT[key], castles_number(value))
+            self.assertEqual(EXPECTED_OUTPUT[key], get_castles_number(value))
```

Comment:
Rename methods to ```get_castles_count_1```, ```get_castles_count_2``` please. :man_facepalming: 
Where is your "sense of beauty"?